### DISCUSSION TITLE
How do I change the status code of http redirection to https from 308 to 301?

### DISCUSSION THREAD

**Discussion by ksnyder** [vercel/community#3972] [Help] [answered]
### Summary

This is sort of a duplicate of [](https://github.com/orgs/vercel/discussions/1286) but since that ticket is marked 'Answered', even though it isn't truly answered, I am creating this one to catch someone's eye.  The 'answer' in that ticket was "you shouldn't need 301, and if you do tell us why" -- there was not actually a solution to the problem of generating a 301.  I have responded in that ticket with a reason why 301 is needed.  Please let us know how to get Vercel to generate a 301.  Thanks.

### Steps to Reproduce

Set up a 301 redirect.  Vercel will respond with a 308 instead.

**Comment by sebastianroming** [comment, marked as answer]
Hi @ksnyder, 

Vercel will always forward HTTP traffic to HTTPS with a `308` status code ([see our documentation](https://vercel.com/docs/security/encryption#)). 
It's not possible to change the status code to `301` in this case. 

The main reason for this is that the client must repeat the same request method (`GET`, `POST`) on the target location when redirecting with a `308` status code, whereas this is not necessarily the case for a `301` status code (for this status code it's allowed to change the request method from `POST` to `GET`). This may lead to unintended behavior, which is prevented by using the `308` status code.

An excerpt from the [RFC 7238](https://www.rfc-editor.org/rfc/rfc7238):

> Note: This status code is similar to 301 (Moved Permanently)
>       ([[RFC7231], Section 6.4.2](https://www.rfc-editor.org/rfc/rfc7231#section-6.4.2)), except that it does not allow changing
>       the request method from POST to GET.

**Reply by ksnyder** [reply]
How do you handle domain moves with Google search then?  They require a 301.  And to test if the old domain is redirecting properly, they call it with http.  In the [original ticket](https://github.com/orgs/vercel/discussions/1286) this one refers to I have posted the Google error message.

**Reply by bernardogarza** [reply]
In other words, it is impossible to change the address in Google Search Console when the site is hosted on Vercel?
